### PR TITLE
More explicit typing for PendingStateManager.savedOps

### PR DESCRIPTION
**Pull request by markfields**
## Description

Reading through this code, I was uncertain when `IPendingMessage.sequenceNumber` should/must be present or not. I found that it's actually very unambiguous in practice - it's only used after a local message is processed and moved to the `savedOps` array.  So I updated the typing to reflect that.

Got to remove a runtime assert now that TypeScript is helping.

### CODE REVIEW

File: packages/runtime/container-runtime/src/pendingStateManager.ts [hunk 1]
```diff
@@ -46,6 +45,23 @@ export interface IPendingMessage {
 	};
 }
 
+/**
+ * IPendingMessage but without localOpMetadata, which isn't suitable for stashing
```
Comment (rajatch-ff) [inline packages/runtime/container-runtime/src/pendingStateManager.ts:49]: > which isn't suitable for stashing

but the name is `IMessageToStash`. Sorry am confused!

File: packages/runtime/container-runtime/src/pendingStateManager.ts [hunk 2]
```diff
@@ -126,7 +150,7 @@ export class PendingStateManager implements IDisposable {
 	/**
 	 * Sequenced local ops that are saved when stashing since pending ops may depend on them
 	 */
-	private savedOps: IPendingMessage[] = [];
+	private savedOps: ISavedMessage[] = [];
```
Comment (rajatch-ff) [inline packages/runtime/container-runtime/src/pendingStateManager.ts:153]: This comment is for line 148
> private readonly initialMessages = new Deque < IPendingMessageFromStash >();

Few doubts
1. These messages from Stash will never have a sequence number?
2. Should IMessageToStash generically represent messages from/to stash?

File: packages/runtime/container-runtime/src/pendingStateManager.ts [hunk 3]
```diff
@@ -46,6 +45,23 @@ export interface IPendingMessage {
...
+ */
+export interface IMessageToStash extends IPendingMessage {
+	localOpMetadata: undefined;
+}
+
+/**
+ * This represents a message that has been sequenced by the service and processed by this client (aka saved).
+ * It will be included in pending local state, since it may need to be replayed locally on rehydrate.
+ */
+interface ISavedMessage extends IMessageToStash {
```
Comment (rajatch-ff) [inline packages/runtime/container-runtime/src/pendingStateManager.ts:60]: should `ISavedMessage` extend `IPendingMessage`

File: packages/runtime/container-runtime/src/pendingStateManager.ts [hunk 4]
```diff
@@ -101,13 +117,21 @@ function buildPendingMessageContent(
 	return JSON.stringify({ type, contents, compatDetails });
 }
 
-function withoutLocalOpMetadata(message: IPendingMessage): IPendingMessage {
+export function toStash(message: IPendingMessage): IMessageToStash {
```
Comment (rajatch-ff) [inline packages/runtime/container-runtime/src/pendingStateManager.ts:120]: `toStash` doesn't sound as appropriate as `toSavedMessage` to describe a type of Message. `toStashMessage`, `toMessageToStash` or `toUnsavedMessage` or something may help convey the intent better

File: packages/runtime/container-runtime/src/pendingStateManager.ts [hunk 5]
```diff
@@ -33,7 +33,6 @@ export interface IPendingMessage {
 	content: string;
```
Comment (rajatch-ff) [inline packages/runtime/container-runtime/src/pendingStateManager.ts:33]: What all states can an IPendingMessage be in ?
1. Submitted
2. Acked by Server
3. Not Submitted - i.e. IMessageToStash?
4. Sequence - i.e. ISavedMessage?

**Comment by markfields** [comment]
Keeping it alive a little longer...

**Pull request closed without merging**